src/target/stm32f4.c

```diff
@@ -207,34 +207,37 @@ static void stm32f4_detach(target *t)
 
 bool stm32f4_probe(target *t)
 {
-	if (t->part_id == ID_STM32F20X) {
+	uint16_t stored_idcode = t->idcode;
+	t->idcode = target_mem_read32(t, DBGMCU_IDCODE) & 0xFFF;
+
+	if (t->idcode == ID_STM32F20X) {
 		/* F405 revision A have a wrong IDCODE, use ARM_CPUID to make the
-		 * distinction with F205. Revision is also wrong (0x2000 instead
-		 * of 0x1000). See F40x/F41x errata. */
+		* distinction with F205. Revision is also wrong (0x2000 instead
```

Comment:
This reformatting is incorrect - those spaces weren't wrong and clang-format would not have removed them, please don't do that.